dags/app/user_profile/udf.py

```diff
@@ -0,0 +1,327 @@
+            organization,
+            job_title,
+            country_or_region,
+            age_range,
+            gender
+        FROM `{table}`
+        """
+            for table in SOURCE_TABLES
+        ]
+    )
+
+    query = f"""
+            MERGE {USER_PROFILE_TABLE} AS tgt
+            USING (
+                SELECT *
+                FROM (
+                    {union_all_sql}
+                ) 
+                WHERE EXTRACT(YEAR FROM PARSE_DATE('%F', paid_date)) IS NOT NULL
+            ) AS src
+            ON tgt.email = src.email AND tgt.year = EXTRACT(YEAR FROM PARSE_DATE('%F', paid_date))
+            WHEN MATCHED THEN
+                UPDATE SET
+                    tgt.organization = src.organization,
+                    tgt.job_title = src.job_title,
+                    tgt.country_or_region = src.country_or_region,
+                    tgt.age_range = src.age_range,
+                    tgt.gender = src.gender,
+                    tgt.update_time = CURRENT_TIMESTAMP()
+            WHEN NOT MATCHED THEN
+                INSERT (year, email, organization, job_title, country_or_region, age_range, gender, update_time)
+                VALUES (
+                    EXTRACT(YEAR FROM PARSE_DATE('%F', paid_date)),
+                    src.email,
+                    src.organization,
+                    src.job_title,
+                    src.country_or_region,
+                    src.age_range,
+                    src.gender,
+                    CURRENT_TIMESTAMP()
+                )
+            """
+    job = client.query(query)
+    job.result()
+
+
+def get_task_config(task_type: str) -> str:
+    if task_type == "organization":
+        return ORG_REQUIREMENT
+    elif task_type == "job_title":
+        return JOB_REQUIREMENT
+    raise ValueError("不支援的工作類型。請選擇 'organization' 或 'job_title'。")
+
+
+def get_gemini_api_key() -> str:
+    GEMINI_API_KEY = Variable.get("GOOGLE_GEMINI_KEY")
+    if not GEMINI_API_KEY:
+        raise ValueError("GEMINI_API_KEY 未設定")
+    return GEMINI_API_KEY
+
+
+def read_kktix_ticket_user_profile(task_type: str) -> list[str]:
+    client = bigquery.Client()
+    try:
+        query = f"""
+            SELECT DISTINCT {task_type}
+            FROM `{USER_PROFILE_TABLE}`
+            WHERE {task_type} IS NOT NULL
+            AND update_time > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 DAY)
+        """
+        rows = client.query(query).result()  # 等 query 完成
+        # 將每一 row 的 task_type 放到 list
+        data_list = [row[task_type] for row in rows]
+    except NotFound as e:
+        logging.error(f"找不到資料表: {e}")
+        raise
+    except BadRequest as e:
+        logging.error(f"SQL 語法錯誤或欄位不存在: {e}")
+        raise
+    except Forbidden as e:
+        logging.error(f"BigQuery 權限不足: {e}")
+        raise
+    except GoogleAPICallError as e:
+        logging.error(f"BigQuery API 呼叫錯誤: {e}")
+        raise
+    except Exception:
+        logging.exception("未知錯誤")
+        raise
```

Comment:
```suggestion
    except (NotFound, BadRequest, Forbidden, GoogleAPICallError) as err:
        if isinstance(err, NotFound):
            err_msg = "找不到資料表: "
        elif isinstance(err, BadRequest):
            err_msg = "SQL 語法錯誤或欄位不存在: "
        elif isinstance(err, BadRequest):
            err_msg = "BigQuery 權限不足: "
        elif isinstance(err, GoogleAPICallError):
            err_msg = "BigQuery API 呼叫錯誤: "  

        logging.exception(err_msg)
        raise
```

It might be better this way 🤔 unknown error does not provider more information so I'd suggest keep it as it is